### DISCUSSION TITLE
FYI about composer timeout

### DISCUSSION THREAD

**Discussion by matks** [composer/composer#12006] [General]
Hello, I just wanted to share some informations that might be useful

On [PrestaShop](https://github.com/PrestaShop/prestaShop/) repository we use Composer. In the last weeks we started noticing a lot of GitHub Action jobs failing randomly. After investigation it was found the issue was that we were starting a docker container in our CI that was running `composer install` and that step would timeout.

The problem started some weeks ago without clear root cause (we did not modify the CI at this time) on our latest branch. Old patch branches are OK.

After multiple attempts we found one configuration which seems to improve a lot our CI success rate: https://github.com/PrestaShop/PrestaShop/pull/36298/

I share this in case someone would encounter the same issues. I hope it'll help.

We still dont know today why it was working fine some weeks ago and then started deteriorating. Network change on packagist? Change in GitHub Action implementation?

Have a very nice day

**Comment by Seldaek** [comment]
What kind of timeouts are you seeing? git clone timing out or http downloads?

**Reply by matks** [reply]
It was hard to be sure because the only logs we had were the ones from GitHub Action, and we launch a docker image inside GitHub Actions so we the output is even more filtered. (and obviously impossible to reproduce outside of GitHub Action environment)

Most likely http downloads

The solution we found was to perform `composer install` outside of the container, then launch the container. We guess the latency of having a container performing network calls through github actions through the real network was a little too much.

**Reply by Seldaek** [reply]
Ok just curious because you set the COMPOSER_PROCESS_TIMEOUT only applies to git clones really. COMPOSER_IPRESOLVE=4 would disable IPv6 so that definitely might help the HTTP front if there are some new IPv6 routing issues between your servers and ours.
